[choowengyan] iP

choowengyan opened the pull request:
### Whisper

> Whisper is a personal chat that allows you to keep track of all your tasks on your fingertip! 

_Why Whisper?_ 👀
- easy to use 😍
- text-based
- internet free! 

Below are some **features**: 
- [X] Add Todo
- [X] Add Deadline
- [X] Add Event
- [X] Delete Tasks
- [X] Find Tasks
- [ ] more upcoming features! 🔥

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

choowengyan committed:
added welcome and farewell message

choowengyan committed:
echo command and exit func

choowengyan committed:
add ability to store tasks and show list

choowengyan committed:
add task status & task class

choowengyan committed:
added todos, events, deadlines functions

choowengyan committed:
unit testing updates

choowengyan committed:
updates on unit testing script

choowengyan committed:
exception handling

choowengyan committed:
handle delete & exceptions

choowengyan committed:
changed task category using enums

choowengyan committed:
Add save function to the class

choowengyan committed:
Add whipster.txt

choowengyan committed:
Update whipster.txt

YuSoonZ commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,99 @@
+import java.util.*;
```
Need to add package for the class

choowengyan replied:
will edit on this! thanks

YuSoonZ commented on src/main/java/Whisper.java:
```diff
@@ -0,0 +1,238 @@
+import java.util.*;
```
might be good to use specific import packages to fulfil the coding style rule :)

choowengyan replied:
noted! thanks

YuSoonZ reviewed (commented):
Needs a bit of touch up, but overall looks good for improvement :)

choowengyan committed:
Edit save function for event task

choowengyan committed:
Update input file content

choowengyan committed:
Update save function for input file

choowengyan committed:
Add date and times storing function

choowengyan committed:
Update way to read input file

choowengyan committed:
Update new format input file

choowengyan committed:
Handle error and exception

choowengyan committed:
Update text file

dhlee03 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,149 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Task {
```
Clear code👍, but I feel that the class is too long. Consider dividing the task class into different types :)

choowengyan replied:
alright! will work on it, thanks!

dhlee03 commented on src/main/java/WhisperException.java:
```diff
@@ -0,0 +1,45 @@
+import java.util.*;
+
+public class WhisperException extends Exception {
```
Nice Exception handling!

choowengyan replied:
thanks!

dhlee03 reviewed (commented):
Looks good! Keep working on it.

choowengyan committed:
Add Ui Class for OOP purpose

choowengyan committed:
Modify Ui.java

choowengyan committed:
Add Storage.java to load task file

choowengyan committed:
Add user command parsing function

choowengyan committed:
Add command methods

choowengyan committed:
Add task class and methods

choowengyan committed:
Edit methods related to Parser class

choowengyan committed:
Handle exception and errors

choowengyan committed:
Organize into packages

choowengyan committed:
Update whisper.txt

choowengyan committed:
Merge add-grade-support branch

choowengyan committed:
Update whisper.txt

nuyer commented on src/main/java/whisper/Command.java:
```diff
@@ -0,0 +1,147 @@
+                ui.printTaskAsDone(tasks.get(taskIndex));
+                storage.saveFile(tasks);
+            } else {
+                ui.printError("Invalid task number. Please choose between task 1 - " + tasks.size() + "\n");
```
In the MarkCommand and UnmarkCommand classes, the error message is directly printed to the UI. It might be better to throw exceptions and handle them in a higher level of the application, leaving UI classes for UI-related tasks only 👍

choowengyan replied:
Thanks for the suggestion!

nuyer commented on src/main/java/whisper/WhisperException.java:
```diff
@@ -0,0 +1,46 @@
+package whisper;
+import java.util.*;
```
can import specific classes instead of entire packages!

choowengyan replied:
noted on this.

nuyer commented on src/main/java/whisper/WhisperException.java:
```diff
@@ -0,0 +1,46 @@
+                "\n- deadline [description] /by[time]\n- mark [taskID]\n- unmark [taskID]\n- delete [taskID]\n");
+    }
+
+    // tasks has more than 100 task
```
Your comments really help improve readability and understanding, good job!! 💯

choowengyan replied:
thanks!

nuyer commented on src/main/java/whisper/Ui.java:
Well-structured! Easy to Understand the code 🔥

choowengyan replied:
thanks for your suggestions and compliments!

nuyer commented on src/main/java/whisper/Parser.java:
```diff
@@ -0,0 +1,124 @@
+        String[] parts = input.trim().split("\\s+", 2);
+        String command = parts[0].toLowerCase();
+
+        switch(command) {
```
No common mistakes in Switch-case indentation, very good 💫

nuyer commented on src/main/java/whisper/Task.java:
```diff
@@ -0,0 +1,149 @@
+    }
+
+    public enum TaskCategory {
+        T {
```
Can consider making enum names more meaningful (using the exact words Todo, Event, Deadline) 👍

choowengyan replied:
will definitely consider! thanks!

nuyer reviewed (commented):
Overall your code is well-structured and easy to understand. There are few coding standard mistakes previous review already pointed out. Good job, great work!! 🆙

choowengyan committed:
Add unit test for TaskList class

choowengyan committed:
Add Jnit test to Storage class

choowengyan committed:
Add javadoc to methods

choowengyan committed:
Update code according to coding standard

choowengyan committed:
Create find task function

choowengyan committed:
Add javadoc to methods and class

choowengyan committed:
merge branch-A-CodingStandard

choowengyan committed:
Fix merge conflicts

choowengyan committed:
Add java files for GUI creation

choowengyan committed:
Add FXML files for GUI

choowengyan committed:
Add dialog box controllers

choowengyan committed:
Add dependencies for fxml

choowengyan committed:
Add javadoc for Whisper.java

choowengyan committed:
Remove extra lines in Ui.java

choowengyan committed:
Update javadoc for Parser class

choowengyan committed:
Remove commented lines

choowengyan committed:
Add assertions to note important assumptions

In Whisper class, assertion is added to check validity of assumptions
regarding the initialisation of Ui, Storage, and TaskList.

In the Parser class, included assertions to verify assumptions about the
input parsing logic and command creation.

In the Command interface and implementing classes, used assertion to
ensure that critical conditions are met during command execution.

These assertions serve as documentation for developers and help in
identifying potential issues during development or maintenance.

choowengyan committed:
Examines the code and refactor code.

This allows to improve the code quality based on the 'Code Quality' topic, and maximise readabilty.

Codes were being structured properly in for the import statements.

Separating the imports according to their packages is easier to see and checking purpose.

choowengyan committed:
Refactor code to utilise Java Streams in ListCommand

In ListCommand class, utilise `forEach` with a method reference to
iterate over tasks and print each task using the `printTask` method of
the `Ui` class.

The changes leverage Java Streams to make the code more concise and
potentially improve readability. The modifications maintain the
functionality of the commands while adopting a more modern and
expressive coding style.

choowengyan committed:
Merge pull request #2 from choowengyan/branch-A-Assertions

Add assertions to note important assumptions

choowengyan committed:
Merge pull request #3 from choowengyan/branch-A-CodeQuality

Examines the code and refactor code

choowengyan committed:
Merge pull request #4 from choowengyan/branch-A-Streams

Refactor code to utilise Java Streams in ListCommand

choowengyan committed:
Add HelpCommand for in-App guidance

Introduce a new HelpCommand class to provide users with in-App guidance.
The HelpCommand displays a help message to assist users in using the
application effectively.

Additionally, the Ui class has been updated to include a method for
printing the help message. The Parser class has also been modified to
recognize the "help" command and return the corresponding HelpCommand.

This enhancement improves the user experience by offering convenient
access to helpful information within the application.

choowengyan committed:
Add background color to GUI

choowengyan committed:
Add title to window bar

choowengyan committed:
Update user guide on README.md

choowengyan committed:
Update Storage Parser to create Whisper.txt

choowengyan committed:
Remove commented codes in Main.java

choowengyan committed:
Solve overflow content on DialogBox.java

choowengyan committed:
Update command on WhisperException.java
